diff --git a/mailgun.mail.inc b/mailgun.mail.inc
index fc76a55..c1373ae 100644
--- a/mailgun.mail.inc
+++ b/mailgun.mail.inc
@@ -56,12 +56,17 @@ class MailgunMailSystem implements MailSystemInterface {
       'html' => $message['body'],
     );
 
-    // Add the CC and BCC fields if not empty.
-    if (!empty($message['params']['cc'])) {
-      $mailgun_message['cc'] = $message['params']['cc'];
+    // Add CC, BCC and Reply-To fields if not empty.
+    $headers = array_change_key_case($message['headers']);
+
+    if (!empty($headers['cc'])) {
+      $mailgun_message['cc'] = $headers['cc'];
+    }
+    if (!empty($headers['bcc'])) {
+      $mailgun_message['bcc'] = $headers['bcc'];
     }
-    if (!empty($message['params']['bcc'])) {
-      $mailgun_message['bcc'] = $message['params']['bcc'];
+    if (!empty($headers['reply-to'])) {
+      $mailgun_message['h:Reply-To'] = $headers['reply-to'];
     }
 
     $params = array();
